src/server.c

```diff
@@ -4604,6 +4620,11 @@ int processCommand(client *c) {
      * propagation of DELs due to eviction. */
     if (server.maxmemory && !server.lua_timedout) {
         int out_of_memory = (performEvictions() == EVICT_FAIL);
+
+        /* performEvictions may evict keys, so we need flush pending tracking
+         * invalidation keys */
+        trackingHandlePendingKeys();
```

Comment:
removing this will not break the protocol (unlike invalidation during MGET and EXEC), but it can mean you get an invalidation message after you performed on operation on the key, where in fact this message belonged to the old value of the key before it got evicted.
i.e. your last write wasn't invalidated, but i'll look as if it did.
before this PR, that concern didn't exist, but not it does, and i think it's good we solved it.

@huangzhw correct me if i'm wrong, and please also check if there's a test that fails when we remove this.

Reply:
@oranagra You are right. But there is no test. In fact, I don't know how to test it.

Reply:
I see that makes sense, can we clarify that in the comment?

Reply:
you can test it by putting the server in SIGSTOP (see some tests that do that), and then use a deferring client to write several commands, then wake up the server (SIGCONT) and it'll perform all of these commands in one event loop.

Reply:
I use another method.